### DISCUSSION TITLE
openAMP with external interrupt

### DISCUSSION THREAD

**Discussion by Minerro886** [OpenAMP/open-amp#420] [Q&A] [answered]
When I'm running openAMP, I want to receive an external interrupt on the remote processor (freeRTOS) first, and then send the data obtained from the ISR back to the master (Linux). At this time, how do I need to manage this external interrupt and the ipi interrupt from the master. 
I tried it, when I initialize the GIC, the external interrupt can be responded, but the IPI seems to be gone, so the callback function cannot be called.

**Comment by edmooring** [comment, marked as answer]
What hardware platform are you using? This is system-specific code.

**Reply by Minerro886** [reply]
hello. i am using the vck190 of xilinx

**Reply by edmooring** [reply]
This should have been mostly fixed for Xilinx hardware in https://github.com/Xilinx/embeddedsw/commit/3a68890dfb46928a652a24af883b3832f61c35f3

This fix hasn't propagated to the upstream libmetal repository.
